code/addons/test/src/node/vitest-manager.ts

```diff
@@ -283,6 +297,99 @@ export class VitestManager {
     );
   }
 
+  async runAffectedTestsAfterChange(changedFilePath: string) {
```

Comment:
the method has been moved up above other methods, which is why the diff is massive